pipelines/utils/execute_dbt_model/tasks.py

```diff
@@ -85,59 +233,113 @@ def run_dbt_model(
+            if isinstance(_vars, list):
+                vars_dict = {}
+                for elem in _vars:
+                    vars_dict.update(elem)
+                vars_str = json.dumps(vars_dict)
+            else:
+                if isinstance(_vars, str):
+                    try:
+                        json.loads(_vars.replace("'", '"'))
+                        vars_str = _vars.replace("'", '"')
+                    except json.JSONDecodeError:
+                        vars_str = json.dumps(eval(_vars))
+                else:
+                    vars_str = json.dumps(_vars)
+
+            cli_args.extend(["--vars", vars_str])
+
+        log(f"Executing dbt command: {' '.join(cli_args)}", level="info")
+
+        try:
+            dbt_runner = dbtRunner()
+            result = dbt_runner.invoke(cli_args)
+
+            if hasattr(result, "success"):
+                if result.success:
+                    log(
+                        f"DBT runner reports success for {cmd} command",
+                        level="info",
+                    )
+                else:
+                    log(
+                        f"DBT runner reports failure for {cmd} command",
+                        level="warning",
+                    )
+
+            log_file_path = os.path.join(
+                dbt_repository_path, "logs", "dbt.log"
+            )
+
+            if os.path.exists(log_file_path):
+                log(f"Processing DBT log file: {log_file_path}", level="info")
+
+                logs_df = process_dbt_log_file(log_file_path)
+
+                if not logs_df.empty:
+                    log(
+                        f"Found {len(logs_df)} log entries in log file",
+                        level="info",
+                    )
+
+                    log_summary = log_dbt_from_file(log_file_path)
+
+                    model_status = extract_model_execution_status_from_logs(
+                        logs_df
+                    )
+                    if model_status:
+                        log(
+                            f"Model execution status: {model_status}",
+                            level="info",
+                        )
+
+                    if log_summary["error_count"] > 0:
+                        if log_summary["error_count"] > 0:
+                            raise FAIL(
+                                f"DBT '{cmd}' command failed with {log_summary['error_count']} errors. See logs for details."
+                            )
+                else:
+                    log("No log entries found in log file", level="warning")
+            else:
+                log(
+                    f"DBT log file not found at {log_file_path}",
+                    level="warning",
+                )
+
+                if hasattr(result, "success") and not result.success:
+                    raise FAIL(
+                        f"DBT '{cmd}' command failed and no log file was found."
+                    )
```

Comment:
Você fez uma função para analisar o arquivo de log do dbt, certo?

Testei o método `dbt.invoke` e ele já printa os logs. Aparece no prefect os logs?

Deixando esse comentário porque a função `process_dbt_log_file` não é trivial e o `dbt.log` não é um arquivo de dado estruturado, pode aparecer casos que não foi coberto pela função.

Acho que o resultado mais relevante aqui é `result.success` que é um `bool` indicando se o commando foi executado com sucesso. Tem também `result` que é `bool | VersionedSchema | List[str] | Manifest | RunExecutionResult | None`

Reply:
Você pode apenas printar `result.result` se `result.result` não for verdadeiro

Reply:
o dbt.invoke vem com logs de status, mas não com logs de descrições do erro